File: src/NetWorthCalculator.Web.ReactJs/src/Services/BalanceSheet/Dto/LiabilityDto.ts
import { serializable } from "serializr";
import { LiabilityGroup } from "../../../Models/Liability/LiabilityGroup";
import { PaymentInterval } from "../../../Models/Liability/PaymentInterval";

export default class LiabilityDto {
	@serializable id!: number;

	@serializable amount!: number;

	@serializable description!: string;

	@serializable group!: LiabilityGroup;

	@serializable intervalAmount!: number;

	@serializable order!: number;

	@serializable paymentInterval!: PaymentInterval;
}

File: src/NetWorthCalculator.Web.Service/Controllers/BalanceSheet/Dto/LiabilityDto.cs
using NetWorthCalculator.Entities.Enums;

namespace NetWorthCalculator.Web.Service.Controllers.BalanceSheet.Dto
{
	public class LiabilityDto
	{
		public int Id { get; set; }

		public decimal Amount { get; set; }

		public string Description { get; set; }

		public LiabilityGroup Group { get; set; }

		public decimal IntervalAmount { get; set; }

		public int Order { get; set; }

		public PaymentInterval PaymentInterval { get; set; }
	}
}
